src/services/__tests__/QueueService.int.spec.ts

```diff
@@ -0,0 +1,105 @@
+jest.mock("../../repositories/LeaderboardRepository");
+
+let prisma: PrismaClient;
+let seasonSemester: string;
+let seasonYear: string;
+
+beforeEach(async () => {
+  jest.clearAllMocks();
+});
+
+beforeAll(async () => {
+  seasonSemester = getEnvVariable("season_semester");
+  seasonYear = getEnvVariable("season_year");
+
+  prisma = new PrismaClient();
+  await prisma.$connect();
+  await prisma.leaderboard.deleteMany();
+  await prisma.activeMatch.deleteMany();
+  await prisma.queue.deleteMany();
+  await prisma.ballChaser.deleteMany();
+});
+
+afterEach(async () => {
+  await prisma.leaderboard.deleteMany();
+  await prisma.activeMatch.deleteMany();
+  await prisma.queue.deleteMany();
+  await prisma.ballChaser.deleteMany();
+});
+
+afterAll(async () => {
+  await prisma.$disconnect();
+});
+
+const mockPlayer1 = BallChaserQueueBuilder.single({});
+
+describe("QueueService tests", () => {
+  describe("Joining queue", () => {
+    describe("Not already in queue", () => {
+      it("joins a player to the queue if they are not already in the queue", async () => {
+        const result = await joinQueue(mockPlayer1.id, mockPlayer1.name);
+
+        expect(result).toHaveLength(1);
+      });
+
+      it("Not on leaderboard | joins queue with 100 MMR and queue time 1 hour from now", async () => {
+        const result = await joinQueue(mockPlayer1.id, mockPlayer1.name);
+        let mmr = result[0].mmr;
+        let receivedQueueTime = result[0].queueTime.diffNow().as("minutes");
+        let expectedQueueTime = DateTime.now().plus({ minutes: 60 }).set({ millisecond: 0, second: 0 });
+
+        expect(mmr).toBe(null);
+        expect(Math.round(receivedQueueTime)).toEqual(Math.round(expectedQueueTime.diffNow().as("minutes")));
+      });
+      it("On leaderboard | joins queue with Leaderboard MMR and queue time 1 hour from now", async () => {
+        const mockPlayer2 = LeaderboardBuilder.single();
+        mocked(LeaderboardRepository.getPlayerStats).mockResolvedValue(mockPlayer2);
+        const resultJoin = await joinQueue(mockPlayer2.id, mockPlayer2.name);
+        let receivedQueueTime = resultJoin[0].queueTime.diffNow().as("minutes");
+        let expectedQueueTime = DateTime.now().plus({ minutes: 60 }).set({ millisecond: 0, second: 0 });
+
+        expect(resultJoin[0].mmr).toEqual(mockPlayer2.mmr);
+        expect(resultJoin).toHaveLength(1);
+        expect(Math.round(receivedQueueTime)).toEqual(Math.round(expectedQueueTime.diffNow().as("minutes")));
+      });
+    });
+
+    it("Already in the queue | updates queue time to 1 hour from now", async () => {
+      const firstJoin = await joinQueue(mockPlayer1.id, mockPlayer1.name);
+      let receivedQueueTime1 = firstJoin[0].queueTime.diffNow().as("minutes");
+      let expectedQueueTime1 = DateTime.now().plus({ minutes: 60 }).set({ millisecond: 0, second: 0 });
+      expect(Math.round(receivedQueueTime1)).toEqual(Math.round(expectedQueueTime1.diffNow().as("minutes")));
```

Comment:
Small note, can you please add some new lines in here to make it a bit easier to read. Looking at a wall of text can make everything run together.

Reply:
Resolved.